src/server/controllers/telemetry.js

```diff
@@ -1,18 +1,63 @@
 import Telemetry from '../models/telemetry';
+import Flight from '../models/flight';
 import contentResponse from '../helpers/APIResponse';
 
+const NodeCache = require('node-cache');
+const telemetryCache = new NodeCache({ stdTTL: 120, checkperiod: 125 });
+const TELEMETRY_ERROR = 400;
+
 function getTelemetry(req, res) {
-  const flightNameArray = req.params.flightname.split('-');
-  const callSign = flightNameArray[0];
-  const flightNumber = flightNameArray[1];
-  return Telemetry.findOne({ callSign, flightNumber })
-  .then((telemetry) => {
-    res.json(contentResponse(telemetry));
-  }, (err) => {
+  const flightName = req.params.flightname;
+  return checkTelemetryCache(flightName, res);
+}
+
+function getTelemetryForAssignedDevices(foundFlight) {
+  const devices = foundFlight.deviceIds;
+  return Telemetry.findOne({ deviceId: { $in: devices } });
+}
+
+function checkTelemetryCache(flightName, res) {
+  const onCheckTelemetryCache = handleCheckTelemetryCache(flightName, res);
```

Comment:
I have a lousy but consistent way that I deal with this. Happy to compare notes on Wednesday.